spec.bs

```diff
@@ -435,6 +441,207 @@ Sec-PST-Count: Issuing 3 tokens.
 ```
 </div>
 
+Following are the keys used for the PrivateStateTokenV3PMB protocol encoded in TLS presentation language, consisting of elliptic curve scalars and points based
+on the curve choice (P-384).
+
+```
+opaque ECPoint<1..2^16-1>; // X9.62 Uncompressed point.
+opaque Scalar<Ns>; // big-endian bytestring
+
+struct {
+  // Corresponding to the FALSE private metadata bit (x0, y0).
+  Scalar x0;
+  Scalar y0;
+  // Corresponding to the TRUE private metadata bit (x1, y1).
+  Scalar x1;
+  Scalar y1;
+  // Corresponding to the token validity (x˜, y˜).
+  Scalar xs;
+  Scalar ys;
+} TrustTokenSecretKey;
+
+struct {
+  ECPoint pub0; // Corresponding to the FALSE private metadata bit.
+  ECPoint pub1; // Corresponding to the TRUE private metadata bit.
+  ECPoint pubs; // Corresponding to the token validity check.
+} PrivateStateTokenPublicKey;
+```
+
+We use serialization schemes and hashes from draft 07 of the hash-to-curve specification, [[HTC]].
+
+`Ht(t)` is defined as P384_XMD:SHA-512_SSWU_RO_ with a big-endian bytestring input of `t` and a dst of "PMBTokens Experiment V2 HashT".
+
+`Hs(T, s)` is defined to be P384_XMD:SHA-512_SSWU_RO_ with a bytestring input of `T` with the following content and a dst of "PMBTokens Experiment V2 HashS".
+
+```
+struct {
+  ECPoint t;
+  opaque s[Nn];
+} T;
+```
+
+The [[HTC]] document does not define hash to scalars, so `Hc(x)` is defined to be the output of the hash_to_field function with the following parameters:
+
+* `DST` is "PMBTokens Experiment V2 HashC"
+* `F`, `p`, and `m` are defined according to the finite field `GF(r)`, where `r` is the order of P-384. Note this is a different modulus from `hash_to_field` as used in P384_XMD:SHA-512_SSWU_RO.
+* `L` is 72, derived based on P384_XMD:SHA-512_SSWU_RO_'s security parameter `k` (192), and `p` defined above.
+* `expand_message` uses the corresponding function from P384_XMD:SHA-512_SSWU_RO_.
+
+When used in the DLEQ proof, `Hc` takes the following input.
+
+```
+struct {
+  uint8 label[6] = "DLEQ2\0";
+  ECPoint X;
+  ECPoint T;
+  ECPoint S;
+  ECPoint W;
+  ECPoint K0;
+  ECPoint K1;
+} DLEQInput; // Used for DLEQ proof.
+```
+
+When used in the DLEQOR proof, `Hc` takes the following input.
+
+```
+struct {
+  uint8 label[8] = "DLEQOR2\0";
+  ECPoint X0;
+  ECPoint X1;
+  ECPoint T;
+  ECPoint S;
+  ECPoint W;
+  ECPoint K00;
+  ECPoint K01;
+  ECPoint K10;
+  ECPoint K11;
+} DLEQORInput; // Used for DLEQOR proof.
+```
+
+When used in the DLEQ batching step, `Hc` is called once for each `e_i` output, with the following input. The `index` field contains `i`.
+
+```
+struct {
+  uint8 label[11] = "DLEQ Batch\0";
+  ECPoint pubs;
+  ECPoint pub0;
+  ECPoint pub1;
+  ECPoint T'0;
+  ECPoint S'0;
+  ECPoint Wp0;
+  ECPoint Wsp0;
+  ...
+  ECPoint T'n;
+  ECPoint S'n;
+  ECPoint Wpn;
+  ECPoint Wspn;
+  uint16 index;
+} DLEQBatchInput; // Used for Batch proof.
+```
+
+The `Issue` function corresponds to the **AT.Sig** stage of the [[PMB]] protocol.
```

Comment:
This usage sounds informative to me but @dvorak42 mentioned that this is normative as well.